### PR TITLE
dev: add type annotations to app

### PR DESCRIPTION
**Pull request by maratori**

### CODE REVIEW

File: lona/app.py [hunk 1]
```diff
@@ -72,40 +79,73 @@ def resolve_path(self, path):
...
 
-    def middleware(self, middleware_class=None):
-        def decorator(middleware_class):
+    @overload
+    def middleware(self) -> Callable[[Type], None]: ...
+
+    @overload
+    def middleware(self, middleware_class: Type) -> None: ...
+
+    def middleware(self, middleware_class: Optional[Type] = None) -> \
```
Comment (fscherf) [inline lona/app.py:111]: In cases like this i would rather add ``# NOQA: $ERROR_CODE``

File: lona/app.py [hunk 2]
```diff
@@ -157,7 +207,7 @@ def add_static_file(self, name, string='', path=''):
         )
 
     # server ##################################################################
-    def run(self, **args):
+    def run(self, **args: Any) -> None:  # TODO: change to explicit arguments
```
Comment (fscherf) [inline lona/app.py:210]: Does this mean your pr is not ready, or do you want to add this todo string?

File: lona/app.py (hunk 1 above)
Comment (maratori) [reply lona/app.py:111]: Does your comment relates to overloading or to multiline declaration?

Comment (fscherf) [reply lona/app.py:111]: @maratori Multiline declaration. I don't like the ``\`` syntax

Comment (maratori) [reply lona/app.py:111]: Ok. I'll rebase this after #75 is merged.

Comment (maratori) [reply lona/app.py]: I've decided to not wait for #75.

File: lona/app.py (hunk 2 above)
Comment (maratori) [reply lona/app.py]: Removed comment

File: lona/app.py [hunk 3]
```diff
@@ -72,40 +83,71 @@ def resolve_path(self, path):
...
+            decorator(middleware_class)
+            return None
 
         else:
             return decorator
 
-    def frontend_view(self, view_class=None):
-        def decorator(view_class):
+    @overload
+    def frontend_view(self) -> Callable[[Type[LonaView]], None]: ...
```
Comment (fscherf) [inline lona/app.py:127]: @maratori I don't like the one-line syntax. Could you change it to two separate lines?

File: setup.py [hunk 4]
```diff
@@ -18,6 +18,8 @@
         'aiohttp>=3,<4',
         'jinja2',
         'rlpython',
+        'typing',
+        'typing-extensions',
```
Comment (fscherf) [inline setup.py:22]: @maratori This is not part of the python standard lib?

**Review by fscherf — CHANGES_REQUESTED** [review]

File: lona/app.py (hunk 3 above)
Comment (maratori) [reply lona/app.py]: Sure, fixed.

File: setup.py (hunk 4 above)
Comment (maratori) [reply setup.py:21]: You are right, `typing` is stdlib (I removed it).
But `typing-extensions` isn't.
```
$ docker run --rm python python -c 'import typing_extensions'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'typing_extensions'
```

Comment (fscherf) [reply setup.py:21]: @maratori the documentation of typing-extensions states 

>Typing Extensions – Backported and Experimental Type Hints for Python

We don't need backports, or do we? And should we use experimental features?

Comment (maratori) [reply setup.py:21]: We don't use experimental features, only one backported type `Literal`.
It was introduced in [Python 3.8](https://docs.python.org/3/library/typing.html#typing.Literal).
So we have to use `typing-extensions`, see https://github.com/lona-web-org/lona/pull/69/files#diff-95f7c4beb7cedde530d15bdf9b7c86d405985af7b5fd18ad067728b756f80e7bR11.

Comment (fscherf) [reply setup.py:21]: Ah ok! Thanks for the explanation

**Comment by fscherf** [comment]
@maratori thanks! :)

**Pull request merged**